server/server.go

```diff
@@ -519,12 +520,56 @@ func (server HTTPServer) singleRuleEndpoint(writer http.ResponseWriter, request
 		return
 	}
 
+	if rule.Internal == true {
+		ok := server.checkInternalRulePermissions(writer, request)
+		if ok != true {
+			// handled in function
+			return
+		}
+	}
+
 	err = responses.SendOK(writer, responses.BuildOkResponseWithData("report", rule))
 	if err != nil {
 		log.Error().Err(err).Msg(responseDataError)
 	}
 }
 
+// checkInternalRulePermissions checks if organizations for internal rules are enabled
+// if so, retrieves the org_id from request/token and returns whether that ID is on the list
+// of allowed organizations to access internal rules
+func (server HTTPServer) checkInternalRulePermissions(writer http.ResponseWriter, request *http.Request) bool {
+	if !server.Config.EnableInternalRulesOrganizations || !server.Config.Auth {
+		return true
+	}
+
+	orgAllowed := false
+
+	authToken, err := server.GetAuthToken(request)
+	if err != nil {
+		handleServerError(writer, err)
+		return false
+	}
+
+	requestOrgID := int(authToken.Internal.OrgID)
+
+	log.Info().Msgf("Checking internal rule permissions for Organization ID: %v", requestOrgID)
+	for _, allowedID := range server.Config.InternalRulesOrganizations {
```

Comment:
Btw, how many rules internal rules organizations do we expect?

Reply:
possibly it will be disabled (I mean the filter)

Reply:
(I know it is not the right answer though :)

Reply:
@Sergey1011010 @tisnik exactly, from zero to I dont know :)